bindings/wasm/src/account/storage/traits.rs

```diff
@@ -0,0 +1,292 @@
+    Ok(Some(state.into()))
+  }
+
+  /// Sets a new state for the identity specified by `did`.
+  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> AccountStorageResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_state(did.clone().into(), state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Removes the keys and any state for the identity specified by `did`.
+  async fn purge(&self, did: &IotaDID) -> AccountStorageResult<()> {
+    let promise: Promise = Promise::resolve(&self.purge(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const STORAGE: &'static str = r#"
+/** All methods an object must implement to be used as an account storage. */
+interface Storage {
+  /** Sets the account password.*/
+  setPassword: (encryptionKey: Uint8Array) => Promise<void>;
+
+  /** Write any unsaved changes to disk.*/
+  flushChanges: () => Promise<void>;
+
+  /** Creates a new keypair at the specified `location`, 
+   * and returns its public key as a base58 string.*/
+  keyNew: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Inserts a private key, encoded as a base58 string, at the specified `location`, 
+   * and returns its public key as a base58 string.*/
+  keyInsert: (did: DID, keyLocation: KeyLocation, privateKey: string) => Promise<string>;
+
+  /** Returns `true` if a keypair exists at the specified `location`.*/
+  keyExists: (did: DID, keyLocation: KeyLocation) => Promise<boolean>;
+
+  /** Retrieves the public key, encoded as a base58 string, from the specified `location`.*/
+  keyGet: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Deletes the keypair specified by the given `location`. Nothing happens if the key is not found.*/
+  keyDel: (did: DID, keyLocation: KeyLocation) => Promise<void>;
+
+  /** Signs `data` with the private key at the specified `location`.*/
+  keySign: (did: DID, keyLocation: KeyLocation, data: Uint8Array) => Promise<Signature>;
+
+  /** Returns the chain state of the identity specified by `did`.*/
+  chainState: (did: DID) => Promise<ChainState | undefined | null>;
+
+  /** Set the chain state of the identity specified by `did`.*/
+  setChainState: (did: DID, chainState: ChainState) => Promise<void>;
+
+  /** Returns the state of the identity specified by `did`.*/
+  state: (did: DID) => Promise<IdentityState | undefined | null>;
+
+  /** Sets a new state for the identity specified by `did`.*/
+  setState: (did: DID, identityState: IdentityState) => Promise<void>;
+
+  /** Removes the keys and any state for the identity specified by `did`.*/
+  purge: (did: DID) => Promise<void>;
+}"#;
+
+// from https://github.com/rustwasm/wasm-bindgen/issues/2231#issuecomment-656293288
+#[allow(dead_code)]
+pub fn downcast_js_value<T: FromWasmAbi<Abi = u32>>(
```

Comment:
Maybe make this function private since it's only used within this file (and its use should be strongly discouraged anywhere else).

```suggestion
// from https://github.com/rustwasm/wasm-bindgen/issues/2231#issuecomment-656293288
#[allow(dead_code)]
fn downcast_js_value<T: FromWasmAbi<Abi = u32>>(
```

Reply:
Done in #690